sys/net/link_layer/ieee802154/submac.c

```diff
@@ -337,6 +337,25 @@ int ieee802154_submac_init(ieee802154_submac_t *submac, const network_uint16_t *
         submac->channel_page = CONFIG_IEEE802154_DEFAULT_SUBGHZ_PAGE;
     }
 
+    /* Get supported PHY modes and select one */
+    int phy_modes = ieee802154_radio_get_phy_modes(dev);
+
+    /* Check if configuration provides valid and supported PHY */
+    if (CONFIG_IEEE802154_DEFAULT_PHY_MODE != IEEE802154_PHY_DISABLED &&
+        phy_modes & CONFIG_IEEE802154_DEFAULT_PHY_MODE) {
+        submac->phy_mode = CONFIG_IEEE802154_DEFAULT_PHY_MODE;
+    }
+    else {
+        /* Get first set bit, and use it as the default,
+         * by this order, the priority is defined on the IEEE 802.15.4 PHY
+         * modes definition, first IEEE 802.15.4-2006 PHY modes, then
+         * IEEE 802.15.4g-2012 PHY modes. */
+        uint8_t bit;
+        bitarithm_test_and_clear(phy_modes, &bit);
```

Comment:
I think you rather want to use `bitarithm_lsb()` here - the order in which bits are processed by `bitarithm_test_and_clear()` is not defined.

Reply:
Thanks, solved at ea5ea1913a